src/main/java/com/internet/shop/model/User.java

```diff
@@ -24,21 +25,20 @@ public User(Long id, String name,
         this.password = password;
     }
 
-    public User(String name, String login,
-                String password, Set<Role> roles) {
+    public User(String name, String login, String password, Set<Role> roles, byte[] salt) {
         this.name = name;
         this.login = login;
         this.password = password;
         this.roles = roles;
```

Comment:
![image](https://user-images.githubusercontent.com/50640794/94657489-35acb780-030a-11eb-81bc-1794b653495e.png)
let's modify your constructors like this